bsp/src/mill/bsp/BSP.scala

```diff
@@ -64,7 +64,7 @@ object BSP extends ExternalModule {
     write(
       BspConfigJson(
         name = "mill-bsp",
-        argv = Seq(millPath, "-i", s"${BSP.getClass.getCanonicalName.split("[$]").head}/start"),
+        argv = Seq(millPath, "-i", "--disable-ticker", "--color", "false", s"${BSP.getClass.getCanonicalName.split("[$]").head}/start"),
```

Comment:
The `--disable-ticker` is not necessarily needed to make BSP work, but removes valuable information about what is being run by mill-bsp. At least when I try with IntelliJ IDEA, I can see the ticker in the `Build: Sync` window, and it's nice to see what's going on.

Reply:
Okay, but I think it works due to luck. We want BSP to be able to log "cleanly" without Mill adding onto the output. I think the place we would want to enable the ticker would be in the MillBspLogger here https://github.com/lihaoyi/mill/blob/7530cf2e0b723590eb9febf1a2192ae888a61bcd/bsp/src/mill/bsp/MillBspLogger.scala#L21. As I understand it, that's the part responsible for wrapping task output so it can be sent to the IDE for display.

Reply:
Hm, the `ticker` is used as progress, but has no progress semantic in it's API, so it will always be pure luck to parse it in the `MillBspLogger`. After all, the logger API is available to every target, including user written ones.

Of course, we can keep it disabled, but where am I suspected to see the ticker/progress (which is send to the BSP client in `MillBspLogger`) in the client, esp. IDEA?

Reply:
I think the way it should work is that BSP.start should be given a logger with all decoration disabled. MillBspLogger wraps that logger and enables the ticker (this is not currently possible since the ticker is managed by the root PrintLogger, but this can be changed). 

This will mean that the messages written to T.ctx.log by BSP will be "raw" which is what we want. When BSP runs a task, it will give that task the MillBspLogger, so the logs generated by those tasks will have the ticker. Those logs are sent to the IDE as TaskProgress messages, which I believe will show up in the "Build output" tab in IntelliJ.